### DISCUSSION TITLE
Show possible pp on a choked play

### DISCUSSION THREAD

**Discussion by Zockerboi** [ppy/osu#29396] [Ideas]
So, if you for example miss on a map, instead showing only what an SS would be worth in the result screen, it should also show a "if not choked" pp score. See on the screenshot how i was thinking it:
![Unbenannt](https://github.com/user-attachments/assets/a9627650-1637-4baa-807e-a3fec51499d7)

**Comment by vatei** [comment]
Would be super nice, I hope it fits into the result screen redesign though

**Comment by Joehuu** [comment]
Has been attempted in https://github.com/ppy/osu/pull/25624 but is stuck on some mania edge case.

**Reply by DanielPower** [reply]
It doesn't really seem like it's stuck on a mania edge case - the feature just doesn't make sense for mania's scoring system and wouldn't be shown in mania. It seems like it's moreso stuck on a maintainer being opposed to the idea of including the feature.

I'd really love to see this feature implemented. I find myself frequently alt-tabbing after a play to get this stat from a Discord bot. Wouldn't need to if it was just shown in the results screen.

Edit: Although if combo scaling removal goes through, maybe this feature won't be needed for std in the future either.

**Reply by gowon11** [reply]
To be honest it just seems like the person evaluating it was just super biased against it. Not following through because it would exclude mania, a gamemode where this feature doesn't have a reason to exist, is just super weird.

**Comment by bdach** [comment]
I still don't want this in the game fwiw. I know that people want this but I dunno. This gives me the same air of "what if though" as people who seriously propose deleting NF score penalty. If you want to find out how much pp you could get if you FC or something then I say you should go use a discord bot as people do right now. Or actually hit the play.

**Reply by Zockerboi** [reply]
In my opinion it should be in the game for three points:

- The performance breakdown menu in my opinion has no effective use case, only if you grind for SSes.

- The action for that should be easier to use, as players who focus on PP need to use the discord bot. An "if FC" line in that menu wouldn't really mess with the ui too since it's only one single line.

- As mentioned, something like this would be not noticable due to the size by casual players, but for those who wants it it's a great implementation.

**Reply by bdach** [reply]
> The performance breakdown menu in my opinion has no effective use case, only if you grind for SSes.

If that is the case then you might as well argue for removing it entirely rather than putting more info there.

> The action for that should be easier to use, as players who focus on PP need to use the discord bot. An "if FC" line in that menu wouldn't really mess with the ui too since it's only one single line.

I don't want to "focus on PP" or encourage pp farming any more than it already is.

**Reply by Tenexxt** [reply]
Understandable that you don't want to encourage pp farming, but to be fair people who are farming won't stop just because of this being added, and afaik most farmers are on stable
I you want to focus on making the pp farming community smaller, or just not adding much stuff to it, then there are many other things that could help, like not allowing songs that already have 10 ranked sets have 10 qualified ones, which 8 of the qualified 10 is nominated by the same BN :) Basically, by improving the ranked system you will also get somewhat rid of pp farming, because by fixing this issue people can't just nominate farm and also players won't have their top leaderboard filled with one song, and still getting bonus pp for it
![Screenshot_2024-08-13-02-14-06-400_com microsoft emmx-edit](https://github.com/user-attachments/assets/c99321eb-5fbc-40fe-a9c5-bafc371199e3)

A counter for a FC PP isn't a problem, since there already is a counter for an SS, so adding this is just a QoL change that would help a big group of people, rather than only pp farmers. Also it not being compatible with mania shouldn't be a reason to not do this because every mode has a different scoring system, and it's no one's fault that the system works the way it does for a certain mode.

**Reply by gowon11** [reply]
Also would like to add that this feature seems to contribute to the path that lazer seems to be taking, which is removing the necessity of leaving the game to access things that aren't just playing the game like accessing player profiles, using the beatmap listing like it's on the website, downloading tools to analyzing replays, etc. Adding this would remove the necessity of third party apps to see what could have been achieved on said score if you just grinded a bit more for an FC. I don't really see why this shouldn't be in the game since it's just one aditional line in the performance breakdown.

**Reply by bdach** [reply]
It's as simple as: I resent the idea of the game becoming a "pp farming simulator" and do not want to lean into the demands of that crowd too hard. This proposal feels like too hard.

**Comment by cl8n** [comment]
I think this performance breakdown area should focus on giving you detailed stats about the play that happened, not "what if" scenarios. I'm not completely against having some way to simulate scores in-game -- just by general principle that creating conveniences for popular use-cases is good -- but presenting it here definitely gives me "pp farming simulator" vibe with the same worries bdach mentioned

**Comment by sinc0s** [comment]
I'm going to stray away from the concept of the "pp farming simulator", and also the simple reason that it is convenient (although I still believe this is a good enough argument alone) to propose a different reason why I believe this should be added. This is likely very similar to minisbett's comment: https://github.com/ppy/osu/pull/25624#issuecomment-1849259221

This addition would be a massive motivator for many players. The maximum does not serve well enough in this purpose because the majority of players cannot/will not be striving to SS the map they are playing. This metric is a far better identifier for the player of what they could possibly achieve if they kept trying to improve on the map. Similar to medals, this becomes something which is a strong motivator for the player to keep going, but is far more likely to actually be achievable for them. As much as it is sometimes undesirable, getting PP is the main source of enjoyment for a lot of players, and this would only help.

I also think that the argument for opposing this due to not being in the osu!mania ruleset is strange. It makes perfect sense for this to not exist in osu!mania, consider the fact that the ruleset relies purely on accuracy in performance calculation (to my knowledge). To have this apply to only rulesets that account for misses in calculation seems completely fine.

**Reply by bdach** [reply]
I have an inverse feeling. I could just as well argue that showing this number feels like rewarding the player with seeing a number they didn't actually earn, i.e. demotivating. As in, think "I _could_ hit this much pp if I hadn't choked so I'm not gonna try anymore", i.e. what the kids call "cope" these days.

Sorry but I can't be moved on this I think. I fell a bit into the farming hole for a bit in 2019 and what it did is it ruined my enjoyment of the game for a bit. Only when I caught myself and realised what was happening *and stopped the pp chase* I got it back. You'll have to find other people to approve and merge this sort of change. I speak only for myself in this matter, but I will continue to not be actively interested in merging this feature.

> It makes perfect sense for this to not exist in osu!mania, consider the fact that the ruleset relies purely on accuracy in performance calculation (to my knowledge).

If you know all these details it may "make perfect sense". It won't make perfect sense to someone who's opened the game for the first time in their life.

**Comment by ziv-vy** [comment]
Maybe a better solution that could both satisfy the pp homo sapiens and the pls enjoy game devs would be the ability to right click a note during a replay in order to bring a right click context menu and change its hit judgement.
(standard) If the player clicked but missed the note, it would also highlight the one he'd have gotten for aiming correctly.

**Comment by monochrome22** [comment]
I agree that showing if-FC at results screen is a bit much; it feels like dangling a carrot in front of someone, trying to tempt them to PP farm.  But since if-FC is a popular feature, I think it should still be added in some way.
One solution would be to add it as a skinning option into either BeatmapAttributeText or the PP counter. This would cause it to become less accessible and convenient, which would keep it away from casual players more. Similar to how the accuracy counter has display modes, the PP counter could have display modes for "Current", "Maximum achievable", "For Full Combo", "Maximum achieved".

**Reply by Taneta-nemausernames** [reply]
Doesn't it exist right now? The max pp at the results screen is the if fc

**Reply by monochrome22** [reply]
No, that's if SS.
